#166214086 heroku scripts

Nennyfills opened the pull request:
What does this PR do?
Deploy an application to Heroku

Description of Task to be completed?
- Add build script to the package.json file
- Modify start script to the package.json file
- Add engines version to the package.json file 
- Add Heroku-post-build script
- Create Heroku Procfile file

How should this be manually tested?
N/A

Any background context you want to provide?
N/A

Heroku Link
[hello-books-be-staging](hello-books-be-staging.herokuapp.com)
[hello-books-be](hello-books-be.herokuapp.com)

What are the relevant pivotal tracker stories?
[#166214086](https://www.pivotaltracker.com/story/show/166214086)

Screenshots (if appropriate)
N/A

Ukhu committed:
166214082-task(database):setup Postgresql to use Sequelize ORM

[Finishes #166214082]

Nennyfills committed:
ch(heroku):heroku scripts
add build script
modify start script
create heroku Procfile file
add engines version
[#166214086]

Nennyfills committed:
ch(heroku):heroku scripts
add build script
modify start script
create heroku Procfile file
add engines version
[#166214086]

allebd commented on Procfile:
```diff
@@ -0,0 +1 @@
+web: npm start
```
A newline is required at the end of the file

Nennyfills replied:
done

allebd commented on package.json:
```diff
@@ -4,7 +4,8 @@
     "scripts": {
-        "start": "node server/index.js",
+        "build": "babel ./server -d build",
+        "start": "babel  ./server -d build && node build/index.js",
```
You can use npm run build on the start script instead of the duplicate statements

Nennyfills replied:
The correction has been made please review this again.

Lundii committed:
Merge pull request #11 from KvNGCzA/ch-setup-database-166214082

#166214082 Database setup

KvNGCzA commented on package.json:
```diff
@@ -4,7 +4,8 @@
     "main": "index.js",
     "scripts": {
-        "start": "node server/index.js",
+        "build": "babel ./server -d build",
```
build into a folder called `dist`.

Nennyfills replied:
I have corrected that, please review this again.

KvNGCzA commented on package.json:
```diff
@@ -4,7 +4,8 @@
-        "start": "node server/index.js",
+        "build": "babel ./server -d build",
+        "start": "babel  ./server -d build && node build/index.js",
         "start:dev": "nodemon --exec babel-node server/index.js",
```
Add a heroku-postbuild script also

Lundii reviewed (commented):
You have merge conflicts in your package.json file. Please resolve conflicts.

Nennyfills committed:
ch(update-script):modify package.json file
modify build script
modify start script
add heroku-postbuild script
[#166214086]

Nennyfills committed:
ch(resolve-conflict):merge conflict
merge conflict in package.json
[#166214086]
